Inferences/Induction.cpp

```diff
@@ -817,7 +823,8 @@ void InductionClauseIterator::processLiteral(Clause* premise, Literal* lit)
         resolveClauses(kv.first, ctx, kv.second);
       }
     }
-  } else if (!env.options->inductionGroundOnly() && InductionHelper::isStructInductionOn() && InductionHelper::isNonGroundInductionLiteral(lit)) {
+  } else if (((env.options->questionAnswering() == Options::QuestionAnsweringMode::SYNTHESIS) || !env.options->inductionGroundOnly()) &&
```

Comment:
Hm, shouldn't this be expressed with a constraint on the level of options? If we have QA and induction, then `indgo` should be `off`? Otherwise, this could be even unexpected for the user. What if someone wants to use induction to prove their synthesized program correct but do not want the program to use recursion?

Reply:
Yes, we can change it later. This is a kind of vague part that will hopefully be unveiled in further papers. 🙂